samples/demo-contract-framework/src/main/java/org/eclipse/dataspaceconnector/demo/contracts/DemoContractOfferFrameworkExtension.java

```diff
@@ -34,14 +39,26 @@
         return Set.of(PROVIDES);
     }
 
+    @Override
+    public Set<String> requires() {
+        return Set.of(AssetIndex.FEATURE, "edc:webservice");
+    }
+
     @Override
     public void initialize(final ServiceExtensionContext context) {
         monitor = context.getMonitor();
 
-        final PublicContractOfferFramework contractOfferFramework = new PublicContractOfferFramework();
-
+        var contractOfferFramework = new PublicContractOfferFramework();
         context.registerService(ContractOfferFramework.class, contractOfferFramework);
 
+        var assetIndex = context.getService(AssetIndex.class);
+        var webService = context.getService(WebService.class);
+        var assetIndexWriter = new InMemoryAssetIndexWriter((InMemoryAssetIndex) assetIndex);
```

Comment:
would expext an AssetIndexWriter to be provided as a service - but guess that will be solved by #199

Reply:
Yes, I think we should merge #199 before if possible